nodebuilder/p2p/p2p.go

```diff
@@ -145,7 +146,15 @@ func (m *module) NATStatus(context.Context) (network.Reachability, error) {
 }
 
 func (m *module) BlockPeer(_ context.Context, p peer.ID) error {
-	return m.connGater.BlockPeer(p)
+	if err := m.connGater.BlockPeer(p); err != nil {
+		return err
+	}
+	if m.host != nil {
+		if err := m.host.Network().ClosePeer(p); err != nil {
+			log.Warnf("failed to close connection to blocked peer %s: %v", p, err)
+		}
+	}
```

Comment:
The Host field should always be initialized, so there shouldn’t be a nil check for it. Could you please remove the if m.host != nil condition to reflect this?

Additionally, could you modify TestP2PModule_ConnGater to also test the newly introduced Disconnect call? This will help ensure the new logic is fully tested.